### PR TITLE
[MM-16032] Add system ping endpoint health checks

### PR DESCRIPTION
**Pull request by gabrieljackson**
This change adds the option for additional server health checks
to be performed when the system ping endpoint is hit. An additional
field 'getserverstatus' is required to run the enhanced health
checks to ensure previous default ping behavior is not modified.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-16032

### CODE REVIEW

**Comment by gabrieljackson** [comment]
A note on the database check: this is a read-only check right now. I struggled to find a safe low-impact write check I could perform. If you have any ideas or example code of this then I would be happy to add that in.

File: api4/system.go [hunk 1]
```diff
@@ -41,30 +41,61 @@ func (api *API) InitSystem() {
...
-		m["IosMinVersion"] = reqs.IosMinVersion
+	if *c.App.Config().ServiceSettings.GoroutineHealthThreshold > 0 && actualGoroutines >= *c.App.Config().ServiceSettings.GoroutineHealthThreshold {
+		mlog.Warn(fmt.Sprintf("The number of running goroutines (%v) is over the health threshold (%v)", actualGoroutines, *c.App.Config().ServiceSettings.GoroutineHealthThreshold))
+		s[model.STATUS] = model.STATUS_UNHEALTHY
+	}
+
+	// Enhanced ping health check:
+	// If an extra form value is provided then perform extra health checks for
+	// database and file storage backends.
+	if r.FormValue("getserverstatus") != "" {
```
Comment (jwilander) [inline api4/system.go]: Minor but can we use `get_server_status`? We generally use snake case for our query parameters

File: api4/system.go [hunk 2]
```diff
@@ -41,30 +41,61 @@ func (api *API) InitSystem() {
...
+		dbStatusKey := "database-status"
+		s[dbStatusKey] = model.STATUS_OK
+		_, appErr := c.App.Srv.Store.System().Get()
+		if appErr != nil {
+			mlog.Debug(fmt.Sprintf("Unable to get filestore for ping status: %s", appErr.Error()))
```
Comment (jwilander) [inline api4/system.go]: This debug log should say database not filestore right?

File: api4/system.go [hunk 3]
```diff
@@ -41,30 +41,61 @@ func (api *API) InitSystem() {
...
+	if *c.App.Config().ServiceSettings.GoroutineHealthThreshold > 0 && actualGoroutines >= *c.App.Config().ServiceSettings.GoroutineHealthThreshold {
+		mlog.Warn(fmt.Sprintf("The number of running goroutines (%v) is over the health threshold (%v)", actualGoroutines, *c.App.Config().ServiceSettings.GoroutineHealthThreshold))
+		s[model.STATUS] = model.STATUS_UNHEALTHY
+	}
+
+	// Enhanced ping health check:
+	// If an extra form value is provided then perform extra health checks for
+	// database and file storage backends.
+	if r.FormValue("getserverstatus") != "" {
+		dbStatusKey := "database-status"
```
Comment (jwilander) [inline api4/system.go]: Our JSON response in the API use snake case as well so can we update these to be snake case and then maybe use different ones for the headers if we don't want to use snake case there?

**Pull request merged**